src/win/fs.c

```diff
@@ -533,7 +533,12 @@ void fs__close(uv_fs_t* req) {
   else
     result = 0;
 
-  SET_REQ_RESULT(req, result);
+  if (result == -1) {
```

Comment:
Can you add a comment explaining this special-casing is necessary because `_close()` doesn't touch `_doserrno`?